### ISSUE TITLE
The greeter freezes for multi-display setups

### ISSUE THREAD

**Issue by nicolasdumitru** [pop-os/cosmic-epoch#1407]
I am testing COSMIC on my laptop and I have issues logging into the desktop environment when my external is plugged in. Mainly, the login screen is frozen from the very beginning and I cannot type in my password if I have my external monitor plugged in. When no external monitors are plugged in, the greeter more or less works.

In my attempts to fix this, I have tried deleting `~/.local/state/cosmic-comp/outputs.ron` to no avail. I have also tried to disable variable refresh rates on both of my displays; this enabled me to log in... twice. I decided, however, to save the "working" outputs.ron file, switch vrr back on to test it. It has not worked since; I attempted to reproduce the desired behavior by applying the "fix" two more times and it failed.
This is my [outputs.ron](https://github.com/user-attachments/files/18266486/outputs.ron.txt).
The exact version of COSMIC that I tested can be pinpointed by examining my [NixOS config at this particular commit](https://github.com/nicolasdumitru/nixos-config/tree/0ea6107bc06ee2610db6b331a2b6012a992ae1c2), and, in particular, the [flake.lock file](https://github.com/nicolasdumitru/nixos-config/blob/0ea6107bc06ee2610db6b331a2b6012a992ae1c2/flake.lock). I have an Intel CPU and a dedicated Nvidia GPU (this may or may not be relevant). Disabling the Nvidia driver in my config and rebuilding the system does not seem to fix the issue.

**Comment by AlternateIf** [comment]
don't think this is cpu or gpu related. have the same issue on my dual screen desktop. when i start it with only the main one being turned on it works. (after login enabling second monitor works too)

but when i have both monitors enabled from the beginning the login is frozen

**Comment by nicolasdumitru** [comment]
@AlternateIf I don't think it's necessarily hardware related either. In fact, I probably should have been clearer about this, but still, I thought that it might be worth mentioning, just in case.

**Comment by nicolasdumitru** [comment]
Indeed, it seems like it is, in fact, the same issue. I'd be willing to close it as a duplicate, but I'd like a confirmation before taking action. Can you check to see if you can reproduce it on your Nvidia desktop with the version indicated in my issue description? This would be a strong indicator that it's an Nvidia problem. Another reason to believe this is that I first started to experience the issue when the Nvidia driver was updated in nixpkgs. Also, what Nvidia driver version have you managed to reproduce this bug with?

**Comment by GeorgEchterling** [comment]
My Desktop has an AMD GPU, so it's not an Nvidia problem.

**Comment by nicolasdumitru** [comment]
I misread that, my bad

**Comment by nicolasdumitru** [comment]
I've resorted to using SDDM for the moment. The desktop environment itself is working fine and the greeter is the only broken critical component for me. I'd like to also keep testing the greeter, but I will be unable to do so until this issue and pop-os/cosmic-greeter#93 are properly addressed.

**Comment by GeorgEchterling** [comment]
I've swapped the DisplayPort connectors on my monitors, such that my left display is now initialized first (both for the LUKS password screen and for the Greeter) and is shown as display 0 in the settings app. I've rebooted 4 times since then, and the issue only occurred once. The one time it occurred, the LUKS password screen was using a 4:3 resolution, instead of the usual 16:9. Maybe this issue occurs when the leftmost screen's resolution is changed while the Greeter is running/starting?

**Comment by mmstick** [comment]
If you're on Pop!_OS, you could try the Linux 6.12 staging branch. It fixed this issue on my AMDGPU system.

**Comment by GeorgEchterling** [comment]
Using the Linux 6.12 staging branch did not resolve the issue on my system.

```
$ uname -a
Linux rysd-pop 6.12.10-76061203-generic #202412060638~1737460284~24.04~8417cb5 SMP PREEMPT_DYNAMIC Tue J x86_64 x86_64 x86_64 GNU/Linux
```

**Comment by rafaelcga** [comment]
This issue also occurs on a dual display Arch system using kernel 6.12.10 and cosmic-greeter 1.0.0.alpha.5.1-1. Workaround is logging through TTY and executing start-cosmic, but it is not ideal...

**Comment by m-bonanno** [comment]
👍🏻

**Comment by kskarthik** [comment]
I too have the same issue:

version: cosmic alpha6

OS: opensuse tumbleweed

```
 uname -a
Linux suse 6.13.5-1-default #1 SMP PREEMPT_DYNAMIC Fri Feb 28 07:06:44 UTC 2025 (ff9b7ff) x86_64 x86_64 x86_64 GNU/Linux

**Comment by AlternateIf** [comment]
seems to work for me now after recent updates

**Comment by leviport** [comment]
Me as well

**Comment by vitumenezes** [comment]
I've been having this issue for a few months now and I've only just started researching it. First, my internal monitor crashed, and after a while, my external monitor (Dell AW2724DM) crashed too. I have an Acer Nitro (2022):

- AMD® Ryzen 7 4800h with Radeon Graphics ×16
- NVIDIA GeForce GTX 1650/PCIe/SSE2

I tried using just my external monitor and it worked, but this is just a workaround.

```
uname -a
Linux pop-os 6.9.3-76060903-generic #202405300957~1732141768~22.04~f2697e1 SMP PREEMPT_DYNAMIC Wed N x86_64 x86_64 x86_64 GNU/Linux
```


> UPDATE: Some updates were made and for now its working. (since May)
```
uname -a
Linux pop-os 6.12.10-76061203-generic #202412060638~1743109366~22.04~1fce33b SMP PREEMPT_DYNAMIC Thu M x86_64 x86_64 x86_64 GNU/Linux
```

**Comment by aryphy** [comment]
Having the same issue

- Intel i5 12600K
- NVIDIA GeForce RTX 4070 Ti

> uname -a
> Linux pop-os 6.12.10-76061203-generic #202412060638~1748542637~24.04~3446b29 SMP PREEMPT_DYNAMIC Thu M x86_64 x86_64 x86_64 GNU/Linux
